### PR TITLE
Simplified environment variable settings, AIRFLOW_HOME is derived from AIRFLOW_CONFIG.

### PR DESCRIPTION
**Pull request by spaceshipoperator**
Comment at line 224 (original) suggests logic at line 236 may be in err; refactor intended to make more consise (less code).

(this is my first github pull-request, I'm learning, please advise if I erred in contribution)

### CODE REVIEW

File: airflow/configuration.py [hunk 1]
```diff
@@ -221,24 +221,15 @@ def mkdir_p(path):
...
-"~/airflow" and "~/airflow/airflow.cfg" respectively as defaults.
+The environment variable AIRFLOW_CONFIG is a complete path (optionally prefixed with '~/' to indicate user home) along with the file name for the Airflow configuration file, if the file does not exist at this location one will be generated with default settings.  AIRFLOW_HOME is determined by the location of AIRFLOW_CONFIG which defaults to:
+    AIRFFLOW_CONFIG="~/airflow/airflow.cfg"
 """
 
-if 'AIRFLOW_HOME' not in os.environ:
-    AIRFLOW_HOME = os.path.expanduser('~/airflow')
-else:
-    AIRFLOW_HOME = os.path.expanduser(os.environ['AIRFLOW_HOME'])
+AIRFLOW_CONFIG = os.environ.get("AIRFLOW_CONFIG", "~/airflow/airflow.cfg")
```
Comment (mistercrunch) [inline airflow/configuration.py]: I think we still need the `os.path.expanduser` somewhere in here to transform the `~` into an actual path

**Comment by mistercrunch** [comment]
I think you are now inferring `AIRFLOW_HOME` based on `AIRFLOW_CONFIG`, where I think our production setup sets `AIRFLOW_HOME` and the `AIRFLOW_CONFIG` is inferred from the home... all of this is a bit confusing but I don't want to break the current setup for anyone.

File: airflow/configuration.py (hunk 1 above)
Comment (spaceshipoperator) [inline airflow/configuration.py]: agreed (hasty oversight, I thought I had that...apologies)

**Comment by spaceshipoperator** [comment]
You are correct, in addition to making the code here a little easier to follow, the code modification I proposed would for the user (and programmer) to maintain two separate environment variables when one could be determined by the other. I thought this would be a simpler approach (easier for me to follow anyway).  Having said that, I fully acknowledge your concern about impacting existing implementations/deployments, also I recognize that my approach does impose some limit to flexibility.  I'll spend a few minutes with it and see if I can find a sweet spot...(also, I'm still learning github pull requests so if I don't update/notify properly, please point me in the right direction)

File: run_unit_tests.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export AIRFLOW_HOME=${AIRFLOW_HOME:=~/airflow}
+export AIRFLOW_HOME=${AIRFLOW_HOME:=~/airflow_test}
```
Comment (mistercrunch) [inline run_unit_tests.sh]: This might break the way some of us run our tests...

**Pull request closed without merging**